crates/extensions/c8y_http_proxy/src/actor.rs

```diff
@@ -29,59 +24,20 @@ pub struct C8YHttpProxyActor {
 
 impl C8YHttpProxyActor {
     pub fn new(config: C8YHttpConfig, http: ClientMessageBox<HttpRequest, HttpResult>) -> Self {
-        let end_point = C8yEndPoint::new(
-            &config.c8y_http_host,
-            &config.c8y_mqtt_host,
-            config.proxy.clone(),
-        );
-        C8YHttpProxyActor {
-            config,
-            end_point,
-            http,
-        }
+        let end_point =
+            C8yEndPoint::new(&config.c8y_http_host, &config.c8y_mqtt_host, config.proxy);
+        C8YHttpProxyActor { end_point, http }
     }
 
     pub async fn try_get_internal_id(&mut self, device_id: &str) -> Result<String, C8YRestError> {
         let url_get_id: String = self.end_point.proxy_url_for_internal_id(device_id);
+        let request = HttpRequestBuilder::get(&url_get_id).build()?;
 
-        let mut attempt = 0;
-        loop {
-            attempt += 1;
-            let request = HttpRequestBuilder::get(&url_get_id).build()?;
-            let endpoint = request.uri().path().to_owned();
-            let method = request.method().to_owned();
-
-            match self.http.await_response(request).await? {
-                Ok(response) => match response.status() {
-                    StatusCode::OK => {
-                        let internal_id_response: InternalIdResponse = response.json().await?;
-                        let internal_id = internal_id_response.id();
-
-                        return Ok(internal_id);
-                    }
-                    StatusCode::NOT_FOUND | StatusCode::UNAUTHORIZED | StatusCode::FORBIDDEN => {
-                        if attempt > 3 {
-                            error!("Failed to fetch internal id for {device_id} even after multiple attempts");
-                            response.error_for_status()?;
-                        }
-                        info!("Re-fetching internal id for {device_id}, attempt: {attempt}");
-                        sleep(self.config.retry_interval).await;
-                        continue;
-                    }
-                    code => {
-                        return Err(C8YRestError::FromHttpError(
-                            tedge_http_ext::HttpError::HttpStatusError {
-                                code,
-                                endpoint,
-                                method,
-                            },
-                        ))
-                    }
-                },
-
-                Err(e) => return Err(C8YRestError::FromHttpError(e)),
-            }
-        }
+        let http_result = self.http.await_response(request).await?;
+        let http_response = http_result.error_for_status()?;
+        let internal_id_response: InternalIdResponse = http_response.json().await?;
+        let internal_id = internal_id_response.id();
+        Ok(internal_id)
```

Comment:
Ignore that comment. That internal id lookup was valid only when we were eagerly fetching the internal id of the main device during the startup itself (in the `C8yHttpActor::init()`). Now that we don't have that `init` logic and internal id fetched always on-demand, this shouldn't be an issue.

Reply:
Yes, this was one of the intent, however the removed piece of code was useless in case of internet disruptions: TCP and TLS errors breaking the retry loop.

The only place where a retry might is useful is when publishing the software list on start. But this should be done at a different level - in the c8y mapper and not blindly at the HTTP level.